x-pack/plugins/apm/server/routes/alerts/action_variables.ts

```diff
@@ -71,7 +71,7 @@ export const apmActionVariables = {
       'xpack.apm.alerts.action_variables.transactionName',
       { defaultMessage: 'The transaction name the alert is created for' }
     ),
-    name: 'transactionName' as const,
+    name: 'transaction.name' as const,
```

Comment:
I suggested using the ECS field names, like you did here, but I'm getting second thoughts now that I see we've camelcased the other params (`transactionType` and `serviceName`).

@dgieselaar any thoughts?

Reply:
if anything we should keep it consistent. I'm honestly not sure whether dots in field names are well supported in the templating engine used by the alerting framework.

Reply:
It is possible to use dot separated fields in the action template.

<img width="575" alt="Screenshot 2023-04-25 at 12 04 38" src="https://user-images.githubusercontent.com/69037875/234244439-6c040729-bb9b-4dc4-8fc4-1b239bf8ee38.png">

Reply:
But aren't we setting `context['error.grouping_name']` rather than `context.error.grouping_name`?

Reply:
yes, we are setting `error.grouping_name` in the `context`. So the way to access it in template is using `context.error.grouping_name`.

Reply:
Is the alerting framework flattening fields? E.g., there's no difference between `{ 'error.grouping_name': .... }` and `{ error: { grouping_name: ... } }`?

Reply:
Hmm that's not great, I would rather not depend on it here because it feels a bit hackish, but as long as we have tests to cover it (ie, the template gets rendered with the right variables) I don't mind.

Reply:
I don't mind changing the names. We should decide on naming now because we cannot modify it in next version or it will break backward compatibility.

@sqren should I keep `transaction.name` or change to `transactionName`? (same for error.grouing_key, error.grouping_name)

Reply:
Yes, let's change back to camel cased (`transactionName`). Sorry for the noise